evalml/automl/engine/sequential_engine.py

```diff
@@ -20,7 +20,8 @@ def evaluate_batch(self, pipelines):
         while self._should_continue_callback() and index < len(pipelines):
             pipeline = pipelines[index]
             self._pre_evaluation_callback(pipeline)
-            evaluation_result = EngineBase.train_and_score_pipeline(pipeline, self.automl, self.X_train, self.y_train)
+            evaluation_result = EngineBase.train_and_score_pipeline(pipeline, self.automl, self.X_train, self.y_train,
```

Comment:
I'm not convinced that ensembling related logic falls within the Engine's area of responsibility.  I feel like we should keep the engine as responsible for training, fitting, scoring pipelines that are fed to it via train_and_score_pipeline() and freddy's upcoming fit_batch() and evaluate_batch() functionality.  I'm not sure - I'm super tired.  it just seems weird passing X_ensemble and y_ensemble to train_and_score. @dsherry, what do you think?

Reply:
@chukarsten, do you mean you'd prefer a check like:

```python
# in sequential_engine.py
# while loop over pipeines
if pipeline.model_family == ModelFamily.Ensemble:
     evaluation_result = EngineBase.train_and_score_pipeline(pipeline, self.automl, self.X_ensemble, self.y_ensemble)
else:
     evaluation_result = EngineBase.train_and_score_pipeline(pipeline, self.automl, self.X_train, self.y_train)
```

versus what I currently have? Or do something different altogether?

Reply:
Yea, I think I like that. I think at least keeping it out of train and score pipelines will be helpful.

Reply:
I like that too @bchen1116 !